content/en/tutorials/google-ads-for-games.md

```diff
@@ -0,0 +1,377 @@
+    }, this);
+};
+```
+
+First step is to add the Google SDK integration files which we can do by copying the folder 'Adsense for Games' from the [Flappy Bird project][flappy-bird-ads-demo].
+
+![][copy-afg-folder]
+
+And pasting into our fork of the tutorial project.
+
+![][paste-afg-folder]
+
+Let's take a look inside:
+
+[`afg-intergration.js`][afg-intergration.js] adds Google's SDK to the page and also sets the time between ads being served to the user which we can change if need be. For the tutorial, we have also enabled test mode to show placeholders instead of real ads.
+
+```
+// https://developers.google.com/ad-placement/docs/example
+(function() {
+    var script = document.createElement('script');
+    script.setAttribute('data-ad-client', 'ca-pub-2682356268914970');
+
+    // Comment out to disable test mode
+    script.setAttribute('data-adbreak-test', 'on');
+
+    // Ad frequency is user configuble
+    // https://developers.google.com/ad-placement/docs/ad-rate
+    script.setAttribute('data-ad-frequency-hint', '30s');
+
+    // Comment in if you would like to use a specific channel for ad reporting
+    // https://developers.google.com/ad-placement/docs/advanced-reporting
+    // script.setAttribute('data-ad-channel', 'XXXXXX');
+
+    script.src = 'https://pagead2.googlesyndication.com/pagead/js/adsbygoogle.js';
+    document.head.appendChild(script);
+})();
+```
+
+[`afg-setup.js`][afg-setup.js] is our wrapper to make working with ads a bit easier. It will configure Google's SDK and add some helper functions to use for pausing the game before an ad is shown and resuming afterwards.
+
+This includes:
+- disabling/enabling input
+- muting/unmuting sound
+- setting/resetting time scale
+- disabling/enabling rendering
+
+```
+(function() {
+    var app = pc.Application.getApplication();
+
+    window.adsbygoogle = window.adsbygoogle || [];
+    var afg = {};
+
+    afg.adBreak = window.adConfig = function(o) {adsbygoogle.push(o);};
+    afg.ready = false;
+
+    var onReady = function() {
+        afg.ready = true;
+    };
+
+    // Config the ad setup
+    adConfig({
+        // https://developers.google.com/ad-placement/docs/preload-ads
+        preloadAdBreaks: 'on',
+        // https://developers.google.com/ad-placement/docs/manual-sequence
+        onReady: onReady,
+    });
+
+    // Create help functions to disable input and sound
+    var _mouseTargetElement = null;
+    var _keyboardTargetElement = null;
+    var _touchTargetElement = null;
+    var _beforeAdVolume = null;
+    var _beforeAdTimeScale = null;
+    var _beforeAutoRender = null;
+
+    afg.onBeforeAd = function () {
+        // Currently using private API for this. May need to be updated
+        // if PlayCanvas engine is updated
+        _mouseTargetElement = app.mouse._target;
+        _keyboardTargetElement = app.keyboard._element;
+
+        if (app.touch) {
+            _touchTargetElement = app.touch._element;
+        }
```

Comment:
Not ideal so maybe we can revisit this at some point to avoid messing with engine internals.

Reply:
Yep, it's a bit odd that we have public API that reassigns the target HTML element but not to get the element from the input handlers